### PR TITLE
Code styles for csharp

### PR DESCRIPTION
**Pull request by skorlygin**
Add README.md
Add .editorconfig for csharp
Add deploy script
Add package project

### CODE REVIEW

File: csharp/.editorconfig [hunk 1]
```diff
@@ -0,0 +1,333 @@
...
+
+#==========================
+# Require ConfigureAwait()
+#==========================
+
+# CA2007: Do not directly await a Task
+dotnet_diagnostic.CA2007.severity = error
+
+# RCS1090: Add call to 'ConfigureAwait'
+dotnet_diagnostic.RCS1090.severity = none
```
Comment (skorlygin) [inline csharp/.editorconfig:333]: I also added this part related to async

File: csharp/README.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
+## Deploy
+
+If you decide to publish the next version of the nuget packages, you will need to create a tag with a name like "csharp_###" and push it.
+<br />Example:
+```
+$ git tag csharp_1
+$ git push --tags
+```
```
Comment (skorlygin) [inline csharp/README.md:8]: A few words about how to publish it.

File: csharp/README.md [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+### Current _.editorconfig_
+
+Current _**.editorconfig**_ file configures two analyzer packages:
+<br />Microsoft.CodeAnalysis.CSharp.CodeStyle - [Code-style rules overview - .NET](https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/)
+<br />StyleCop.Analyzers - [StyleCopAnalyzers Status](https://dotnetanalyzers.github.io/StyleCopAnalyzers)
+
+## How to include _.editorconfig_
+
+You will need to install the appropriate version of the _Kaseya.CodeAnalysis_ nuget package into your project and all the following magic should appear by itself after the first build.
+<hr />
```
Comment (skorlygin) [inline csharp/README.md:30]: And a few words about how to use it.

File: .github/workflows/deploy-charp.yml [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+      - name: Publish artifacts
+        run: |
+          dotnet nuget push "./out/*.nupkg" \
+            --api-key $GITHUB_TOKEN \
+            --source https://nuget.pkg.github.com/$REPOSITORY_OWNER
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          REPOSITORY_OWNER: ${{ github.repository_owner }}
+
+  # notification:
```
Comment (aglumova) [inline .github/workflows/deploy-charp.yml:48]: Please remove comments.

File: .github/workflows/deploy-charp.yml [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+              --nologo \
+              --configuration Release \
+              --include-symbols \
+              --include-source \
+              --output ./out/
+          done
+
+      - name: Upload artifact
+        uses: actions/upload-artifact@v2
+        with:
```
Comment (aglumova) [inline .github/workflows/deploy-charp.yml:35]: Could you please add screenshots to `README.md` in the sharp folder how this artifact looks in nuget.

**Review by andrey-kozel — APPROVED** [review]

File: csharp/.editorconfig (hunk 1 above)
Comment (aglumova) [inline csharp/.editorconfig:333]: Please add empty line.

File: csharp/.gitignore [hunk 6]
```diff
@@ -0,0 +1,224 @@
...
+**/wwwroot/*
+
+# SpecFlow specific
+*.feature.cs
+*.feature.xlsx.*
+*.Specs_*.html
+
+#####
+# End of core ignore list, below put you custom 'per project' settings (patterns or path)
+#####
```
Comment (aglumova) [inline csharp/.gitignore:224]: Add empty line at the end for all files.

Comment (skorlygin) [reply csharp/.gitignore]: done

File: csharp/.editorconfig (hunk 1 above)
Comment (skorlygin) [reply csharp/.editorconfig]: done

File: .github/workflows/deploy-charp.yml (hunk 4 above)
Comment (skorlygin) [reply .github/workflows/deploy-charp.yml]: done

File: .github/workflows/deploy-charp.yml (hunk 5 above)
Comment (skorlygin) [reply .github/workflows/deploy-charp.yml:35]: done

**Review by aglumova — APPROVED** [review]

**Pull request merged**